csharp/test/Apache.Arrow.Tests/ArrowStreamReaderTests.cs

```diff
@@ -68,7 +68,7 @@ public async Task Ctor_MemoryPool_AllocatesFromPool(bool shouldLeaveOpen)
                 ArrowStreamReader reader = new ArrowStreamReader(stream, memoryPool, shouldLeaveOpen);
                 reader.ReadNextRecordBatch();
 
-                Assert.Equal(1, memoryPool.Statistics.Allocations);
+                Assert.Equal(2, memoryPool.Statistics.Allocations);
```

Comment:
Memo:
`_allocator` is called when reading dictionary batches and record batches.
The expected value of `memoryPool.Statistics.Allocations` is `2` at this context.

Reply:
I fixed it to use `[Theory]`.